django_airavata/apps/api/serializers.py

```diff
@@ -513,6 +515,13 @@ def get_isInputFileUpload(self, data_product):
         request = self.context['request']
         return user_storage.is_input_file(request, data_product)
 
+    def get_path(self, data_product):
+        """Getter for path field."""
+        if len(data_product.replicaLocations) > 0:
+            return re.sub(r'.*/tmp/[^/]*/', "", user_storage._get_replica_filepath(data_product))
```

Comment:
@machristie Can you help me with a workaround to make this generic

Reply:
Is this still needed? With your most recent commit, only the data product uri is needed to update the file contents.

Reply:
It is because from the frontend, first I take the data product from the data product uri and using the path in it only the requests to UserStoragePathView are done. UserStoragePathView requires the path.

Reply:
Can you point out where that is in the code? Thanks.

Reply:
https://github.com/apache/airavata-django-portal/pull/52/files#diff-c7351d24aefb0b542f4eb165ac38b5271701a7fd111e44ed45614e181a4df817R59

Reply:
Sure. Here's in frontend [1] and backend [2] places where the path is required.

[1] https://github.com/dinukadesilva/airavata-django-portal/blob/AIRAVATA-3310-enabling-file-edit-at-the-experiments/django_airavata/apps/workspace/static/django_airavata_workspace/js/containers/UserStorageContainer.vue#L59

[2] https://github.com/dinukadesilva/airavata-django-portal/blob/AIRAVATA-3310-enabling-file-edit-at-the-experiments/django_airavata/apps/api/views.py#L1542

Reply:
@dinukadesilva those links don't seem to be working any longer, meaning they don't point to the right lines. But I tried removing the `path` field from the DataProductSerializer and I was able to upload an input file and then edit it. So I don't think it is needed any longer. Try removing it and see if it works for you too.

Reply:
@machristie Actually the path is needed in the following two
1. If you go to storage and click on one of the files
2. You go to an application, select a file, right-click on the link and open in a new tab.

It's because the "/api/user-storage/~/$path" endpoint needs the path to be given as a param. Earlier this was taken from the window URL. But now we had to change it since the window URL is not yet valid when it's at the experiments window.

Reply:
Neither of those work for me when I try the code locally.  Conceptually, from the API you should only need to provide the data product uri to edit a file, the path shouldn't be needed. The backend can take care of figuring out where the file is actually located.

Maybe we should meet to figure out a way forward.